File: Two Pointers/167. Two Sum II - Input Array Is Sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        n = len(numbers)
        lo = 0
        hi = n - 1

        while lo < hi:
            summ = numbers[lo] + numbers[hi]
            if summ == target:
                return (lo + 1, hi + 1)
            elif summ > target:
                hi -= 1
            else:
                lo += 1

# Time: O(n)
# Space: O(1)

File: Two Pointers/167. Two Sum II - Input Array Is Sorted/167. Two Sum II - Input Array Is Sorted.js
/**
 * @param {number[]} numbers
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (numbers, target) {
  let l = 0;
  let r = numbers.length - 1;

  while (l < r) {
    const summ = numbers[l] + numbers[r];
    if (summ === target) return [l + 1, r + 1];
    else if (summ > target) r--;
    else l++;
  }
};

// Time: O(n)
// Space: O(1)
